pgmpy/inference/continuous_sampling.py

```diff
@@ -0,0 +1,302 @@
+        """
+        # momentum = N(0, I)
+        momentum = np.matrix(np.reshape(np.random.normal(0, 1, len(theta)), (len(theta), 1)))
+
+        # Take a single step in time
+        theta_bar, momentum_bar = self.discretize_time(self.grad_log_pdf, self.model,
+                                                       theta, momentum, epsilon_app).discretize_time()
+        # Parameters to help in evaluating P(theta, momentum)
+        _, logp = self.grad_log_pdf(theta, self.model).get_gradient_log_pdf()
+        _, logp_bar = self.grad_log_pdf(theta_bar, self.model).get_gradient_log_pdf()
+
+        # acceptance_prob = P(theta_bar, momentum_bar)/ P(theta, momentum)
+        potential_change = logp_bar - logp  # Negative change
+        kinetic_change = 0.5 * np.float(np.dot(momentum_bar.T, momentum_bar) - np.dot(momentum.T, momentum))
+
+        acceptance_prob = np.exp(potential_change - kinetic_change)
+
+        # a = 2I[acceptance_prob] -1
+        a = 2 * (acceptance_prob > 0.5) - 1
+
+        condition = (acceptance_prob ** a) > (2 ** (-a))
+
+        while condition:
+            epsilon_app = (2 ** a) * epsilon_app
+
+            theta_bar, momentum_bar = self.discretize_time(self.grad_log_pdf, self.model,
+                                                           theta, momentum, epsilon_app).discretize_time()
+
+            _, logp = self.grad_log_pdf(theta, self.model).get_gradient_log_pdf()
+            _, logp_bar = self.grad_log_pdf(theta_bar, self.model).get_gradient_log_pdf()
+
+            potential_change = logp_bar - logp
+            kinetic_change = 0.5 * np.float(momentum_bar.T * momentum_bar - momentum.T * momentum)
+
+            acceptance_prob = np.exp(potential_change - kinetic_change)
+
+            condition = (acceptance_prob ** a) > (2 ** (-a))
+
+        return epsilon_app
+
+    def sample(self, theta0, num_adapt, num_samples, epsilon=None):
+        """
+        Method to return samples using Hamiltonian Monte Carlo
+
+        Parameters
+        ----------
+        theta0: A 1d array type object or a row matrix of shape 1 X d
+                or d X 1.(Will be converted to d X 1)
+                Vector representing values of parameter theta, the starting
+                state in markov chain.
+
+        num_adapt: int
+                The number of interations to run the adaptation of epsilon,
+                after Madapt iterations adaptation will be stopped
+
+        num_samples: int
+                     Number of samples to be generated
+
+        epsilon: float , defaults to None
+                 The step size for descrete time method
+                 If None, then will be choosen suitably
+
+        Returns
+        -------
+        list: A list of numpy matrix type objects containing samples
+
+        Examples
+        --------
+        >>>
+
+        References
+        ----------
+        Matthew D. Hoffman, Andrew Gelman, The No-U-Turn Sampler: Adaptively
```

Comment:
@ankurankan No, but it uses algorithm for Hamiltonian Monte Carlo, given in that paper.